packages/infra/lib/hl7-notification-routing-stack/mllp.ts

```diff
@@ -0,0 +1,138 @@
+    });
+
+    const mllpSecurityGroup = new ec2.SecurityGroup(this, "MllpServerSG", {
+      vpc,
+      description: "Security group for Fargate MLLP server",
+      allowAllOutbound: true,
+    });
+
+    mllpSecurityGroup.addIngressRule(
+      ec2.Peer.anyIpv4(),
+      ec2.Port.tcp(MLLP_DEFAULT_PORT),
+      "Allow inbound traffic to MLLP server"
+    );
+
+    const fargate = new ecs_patterns.NetworkLoadBalancedFargateService(this, "MllpServerFargate", {
+      cluster,
+      cpu: fargateCpu,
+      memoryLimitMiB: fargateMemoryLimitMiB,
+      desiredCount: fargateTaskCountMin,
+      taskImageOptions: {
+        image: ecs.ContainerImage.fromDockerImageAsset(dockerImage),
+        containerPort: MLLP_DEFAULT_PORT,
+        containerName: "MllpServer",
+        environment: {
+          NODE_ENV: "production",
+          ENV_TYPE: props.config.environmentType,
+          MLLP_PORT: MLLP_DEFAULT_PORT.toString(),
+        },
+      },
+      listenerPort: MLLP_DEFAULT_PORT,
+      healthCheckGracePeriod: Duration.seconds(60),
+      taskSubnets: {
+        subnetType: ec2.SubnetType.PRIVATE_WITH_EGRESS,
+      },
+      publicLoadBalancer: true,
+      loadBalancer: nlb,
+      securityGroups: [mllpSecurityGroup],
+    });
+
+    fargate.targetGroup.configureHealthCheck({
+      port: MLLP_DEFAULT_PORT.toString(),
+      protocol: elbv2.Protocol.TCP,
+      healthyThresholdCount: 3,
+      unhealthyThresholdCount: 2,
+      timeout: Duration.seconds(10),
+      interval: Duration.seconds(20),
+    });
+
+    const scaling = fargate.service.autoScaleTaskCount({
+      minCapacity: fargateTaskCountMin,
+      maxCapacity: fargateTaskCountMax,
+    });
+
+    scaling.scaleOnCpuUtilization("CpuScaling", {
+      targetUtilizationPercent: 70,
+      scaleInCooldown: Duration.seconds(60),
+      scaleOutCooldown: Duration.seconds(60),
+    });
+
+    scaling.scaleOnMemoryUtilization("MemoryScaling", {
+      targetUtilizationPercent: 70,
+      scaleInCooldown: Duration.seconds(60),
+      scaleOutCooldown: Duration.seconds(60),
+    });
+
+    const connectionsPerTaskMetric = new cloudwatch.MathExpression({
+      expression: "connections / tasks",
+      usingMetrics: {
+        connections: new cloudwatch.Metric({
+          namespace: "AWS/NetworkELB",
+          metricName: "ActiveFlowCount",
+          dimensionsMap: {
+            LoadBalancer: fargate.loadBalancer.loadBalancerFullName,
+          },
+          statistic: "Average",
+          period: Duration.minutes(1),
+        }),
+        tasks: new cloudwatch.Metric({
+          namespace: "ECS/ContainerInsights",
+          metricName: "RunningTaskCount",
+          dimensionsMap: {
+            ClusterName: cluster.clusterName,
+            ServiceName: fargate.service.serviceName,
+          },
+          statistic: "Average",
+          // This only changes when we establish a new HIE connection, so this can be updated infrequently
+          period: Duration.minutes(15),
```

Comment:
Wouldn't this define the minimum time for us to be able to scale out/in? Like, the metric is based on the number of tasks, so if we only update it every 15min and we get a scale out event from load on min 2, we'd need to wait more than 10min until we're healthy - during which time we would keep scaling out?

Non-blocker as we can adjust this once we ship it.

Reply:
My understanding is that this number is only going to substantively change when we integrate with a new HIE, so we don't need to change frequently. I dont think the volume of TCP connections we have against the server is going to change by more than a small handful every few hours.

Reply:
Yeah, I'm def not worried about scaling out things now, we'll see when we ship it and then adjust that based on actual data.

My concern here is that this setup could lead to a situation where the server keeps scaling kinda out of control because of what I mentioned before. Could even disable/comment out this until we're ready to test this behavior on staging.

Reply:
Great, I think I'll just delete this then. We have it set to autoscale on mem and cpu, this is probably overkill anyway.

Reply:
Just deleted for now to simplify things. I think this is probably overcomplicating things.